File: 기초 사고력문제/백준_2526_싸이클/사이클.java
import java.util.*;

public class Main{
    public static void main(String[] args){
        Scanner s = new Scanner(System.in);
        int n = s.nextInt();
        int p = s.nextInt();
        int m = n;
        int[] arr = new int[1001];
        int cnt = 0;

        while(true){
            int i = (n*m) % p;
            if(arr[i] > 2) break;
            arr[i]++;
            m = i;
            if(arr[i] == 2) cnt++;
        }
        System.out.println(cnt);
    }
}

File: 기초 사고력문제/백준_2526_싸이클/사이클.kt
import java.util.*

fun main(args: Array<String>) {
    val s = Scanner(System.`in`)
    val n = s.nextInt()
    val p = s.nextInt()
    var m = n;
    var arr =  IntArray(1001, {0})
    var cnt = 0

    while(true){
        var i = (m*n)%p;
        if(arr[i] > 2) break
        arr[i]++
        m = i
        if(arr[i] == 2) cnt++
    }
    print("${cnt}")

}